syne_tune/optimizer/schedulers/searchers/regularized_evolution.py

```diff
@@ -0,0 +1,138 @@
+
+
+@dataclass
+class PopulationElement:
+    score: int = 0
+    config: dict = None
+
+
+class RegularizedEvolution(SearcherWithRandomSeed):
+    def __init__(
+            self,
+            configspace,
+            metric: str,
+            mode: str = 'min',
+            population_size: int = 100,
+            sample_size: int = 10,
+            points_to_evaluate: Optional[List[Dict]] = None,
+            random_seed: int = None
+    ):
+        """
+        Implements the regularized evolution algorithm proposed by Real et al. The original implementation only
+        considers categorical hyperparameters. For integer and float parameters we sample a new value uniformly
+        at random.
+
+        Real, E., Aggarwal, A., Huang, Y., and Le, Q. V.
+        Regularized Evolution for Image Classifier Architecture Search.
+        In Proceedings of the Conference on Artificial Intelligence (AAAI’19)
+
+        The code is based one the original regularized evolution open-source implementation:
+        https://colab.research.google.com/github/google-research/google-research/blob/master/evolution/regularized_evolution_algorithm/regularized_evolution.ipynb
+
+
+        Parameters
+        ----------
+        configspace: dict
+            Configuration space for trial evaluation function
+        metric : str
+            Name of metric to optimize, key in result's obtained via
+            `on_trial_result`
+        mode : str
+            Mode to use for the metric given, can be 'min' or 'max', default to 'min'.
+        population_size : int
+            Size of the population.
+        sample_size : int
+            Size of the candidate set to obtain a parent for the mutation.
+        random_seed : int
+            Seed for the random number generation. If set to None, use random seed.
+        """
+
+        super(RegularizedEvolution, self).__init__(configspace, metric, points_to_evaluate=points_to_evaluate,
+                                                   random_seed=random_seed)
+        self.mode = mode
+        self.population_size = population_size
+        self.sample_size = sample_size
+        self.population = deque()
+
+    def mutate_config(self, config: Dict) -> Dict:
+
+        child_config = copy.deepcopy(config)
+
+        # pick random hyperparameter and mutate it
+        hypers = []
+        for k, v in self.configspace.items():
+            if isinstance(v, Domain):
+                hypers.append(k)
+        name = self.random_state.choice(hypers)
+        hyperparameter = self.configspace[name]
+
+        if isinstance(hyperparameter, Categorical):
+            # drop current values from potential choices to not sample the same value again
+            choices = [cat for cat in hyperparameter.categories if cat != config[name]]
+            new_value = self.random_state.choice(choices)
+
+        elif isinstance(hyperparameter, Float) or isinstance(hyperparameter, Integer):
```

Comment:
Could we not just say "else:" here?
For example, there are also FiniteRange. In the end, since RandomSearcher works for all Domain instances (which has the sample method), why would this one not?